src/main/java/com/zaxxer/nuprocess/internal/BasePosixProcess.java

```diff
@@ -383,23 +397,32 @@ public void readStdout(int availability)
       try {
          if (availability < 0) {
             outClosed = true;
-            processHandler.onStdout(null);
+            outBuffer.flip();
+            processHandler.onStdout(outBuffer, true);
             return;
          }
          else if (availability == 0) {
```

Comment:
I now worry about this conditional.  If the developer does not consume all of the data, as they were required to in the past, they may get into trouble of the buffer fills completely.  They will never be called again to consume data.